pkg/kyverno/test/test_command.go

```diff
@@ -476,35 +476,74 @@ func buildPolicyResults(engineResponses []*response.EngineResponse, testResults
 				test.Policy = userDefinedPolicyName
 			}
 
-			if test.Policy == policyName && test.Resource == resourceName {
-				var resultsKey string
-				resultsKey = GetResultKeyAccordingToTestResults(userDefinedPolicyNamespace, test.Policy, test.Rule, test.Namespace, test.Kind, test.Resource)
-				if !util.ContainsString(rules, test.Rule) {
+			if test.Resources != nil {
+				if test.Policy == policyName {
+					for _, resource := range test.Resources {
+						if resource == resourceName {
+							var resultsKey string
+							resultsKey = GetResultKeyAccordingToTestResults(userDefinedPolicyNamespace, test.Policy, test.Rule, test.Namespace, test.Kind, resource)
+							if !util.ContainsString(rules, test.Rule) {
+
+								if !util.ContainsString(rules, "autogen-"+test.Rule) {
+									if !util.ContainsString(rules, "autogen-cronjob-"+test.Rule) {
+										result.Result = report.StatusSkip
+									} else {
+										testResults[i].AutoGeneratedRule = "autogen-cronjob"
+										test.Rule = "autogen-cronjob-" + test.Rule
+										resultsKey = GetResultKeyAccordingToTestResults(userDefinedPolicyNamespace, test.Policy, test.Rule, test.Namespace, test.Kind, resource)
+									}
+								} else {
+									testResults[i].AutoGeneratedRule = "autogen"
+									test.Rule = "autogen-" + test.Rule
+									resultsKey = GetResultKeyAccordingToTestResults(userDefinedPolicyNamespace, test.Policy, test.Rule, test.Namespace, test.Kind, resource)
+								}
+
+								if results[resultsKey].Result == "" {
+									result.Result = report.StatusSkip
+									results[resultsKey] = result
+								}
+							}
+
+							patchedResourcePath = append(patchedResourcePath, test.PatchedResource)
+							if _, ok := results[resultsKey]; !ok {
+								results[resultsKey] = result
+							}
 
-					if !util.ContainsString(rules, "autogen-"+test.Rule) {
-						if !util.ContainsString(rules, "autogen-cronjob-"+test.Rule) {
-							result.Result = report.StatusSkip
+						}
+					}
+				}
+			}
+			if test.Resource != "" {
```

Comment:
We should only parse one of `resource` or `resources` and emit a warning while processing resource that it is deprecated and users should move to a resources array instead.